packages/js/src/helpers/migrateNotices.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * Transform contents to adhere to the new dashboard format.
+ *
+ * @param {Array} contents An array with contents to be transformed.
+ *
+ * @returns {Array} The contents to be transformed.
+ */
+function transformContent( contents ) {
+	// Transform the buttons to Yoast buttons.
+	contents.forEach( noticeContent => {
+		if ( noticeContent ) {
+			noticeContent.querySelectorAll( "a.button" ).forEach( button => {
+				button.classList.remove( "button" );
+				button.classList.add( "yst-button" );
+				button.classList.add( "yst-button--primary" );
+				button.classList.add( "yst-mt-4" );
+			} );
+		}
+	} );
+
+	return contents;
+}
+
+/**
+ * Prepares notices to be migrated.
+ *
+ * @param {Array} notices The notices to be prepared for migration.
+ *
+ * @returns {Array} The notices, prepared for migration.
+ */
+function prepareNoticesForMigration( notices ) {
+	const ids = notices.map( notice => notice.id );
+	const headers = notices.map( notice => notice.querySelector( ".yoast-notice-migrated-header" ) );
+	const contents = transformContent( notices.map( notice => notice.querySelector( ".notice-yoast-content" ) ) );
+	const isDismissable = notices.map( notice => notice.classList.contains( "is-dismissible" ) );
+
+	return notices.map( ( notice, index ) => ( {
+		notice: notice,
+		id: ids[ index ],
+		header: headers[ index ],
+		content: contents[ index ],
+		isDismissable: isDismissable[ index ],
+	} ) );
+}
+
+/**
+ * Migrates Yoast notices.
+ *
+ * @returns {Array} The migrated Yoast Notices.
+ */
+export function getMigrateNotices() {
```

Comment:
* The `transformContent` is still being done as a side-effect of the `getMigratingNoticeInfo` - I think this adapts the, currently not yet deleted, DOM contents.
* Reintroducing the `useEffect`, here is a whole article about how to try to prevent using `useEffect`: https://react.dev/learn/you-might-not-need-an-effect -- I think this could be lifted to outside React in `initialize` and then passed to `App` as a prop. No need to do this in React